kratos/solving_strategies/schemes/residual_based_pseudo_static_displacement_scheme.h

```diff
@@ -0,0 +1,346 @@
+            noalias(delta_displacement) = current_displacement - previous_displacement;
+
+            current_velocity = previous_velocity;
+
+            current_acceleration = zero_array;
+        }
+
+        KRATOS_CATCH( "" );
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+protected:
+    ///@}
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected  Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    /**
+     * @brief It adds the dynamic LHS contribution of the elements D*c1 + K
+     * @param rLHSContribution The dynamic contribution for the LHS
+     * @param rD The damping matrix
+     * @param rM The mass matrix
+     * @param rCurrentProcessInfo The current process info instance
+     */
+    void AddDynamicsToLHS(
+        LocalSystemMatrixType& rLHSContribution,
+        LocalSystemMatrixType& rD,
+        LocalSystemMatrixType& rM,
+        ProcessInfo& rCurrentProcessInfo
+        ) override
+    {
+        // Adding  damping contribution
+        if (rD.size1() != 0 && TDenseSpace::TwoNorm(rD) > ZeroTolerance) // if D matrix declared
+            noalias(rLHSContribution) += rD * DerivedBaseType::mNewmark.c1;
+        else if (rM.size1() != 0) {
+            const double beta = rCurrentProcessInfo[mRayleighBeta];
+            noalias(rLHSContribution) += rM * beta * DerivedBaseType::mNewmark.c1;
+        }
+    }
+
+    /**
+     * @brief It adds the dynamic RHS contribution of the elements b - D*v
+     * @param pElement The element to compute
+     * @param RHS_Contribution The dynamic contribution for the RHS
+     * @param D The damping matrix
+     * @param M The mass matrix
+     * @param rCurrentProcessInfo The current process info instance
+     */
+    void AddDynamicsToRHS(
+        Element::Pointer pElement,
+        LocalSystemVectorType& rRHSContribution,
+        LocalSystemMatrixType& rD,
+        LocalSystemMatrixType& rM,
+        ProcessInfo& rCurrentProcessInfo
+        ) override
+    {
+        const std::size_t this_thread = OpenMPUtils::ThisThread();
+
+        // Adding damping contribution
+        if (rD.size1() != 0 && TDenseSpace::TwoNorm(rD) > ZeroTolerance) {
+            pElement->GetFirstDerivativesVector(DerivedBaseType::mVector.v[this_thread], 0);
+            noalias(rRHSContribution) -= prod(rD, DerivedBaseType::mVector.v[this_thread]);
+        } else if (rM.size1() != 0) {
+            const double beta = rCurrentProcessInfo[mRayleighBeta];
+            pElement->GetFirstDerivativesVector(DerivedBaseType::mVector.v[this_thread], 0);
+            noalias(rRHSContribution) -= prod(rM * beta, DerivedBaseType::mVector.v[this_thread]);
```

Comment:
put multiplication by beta outside of the prod (less calculations will be done)

Reply:
Done